app/services/bulk_punches_service.rb

```diff
@@ -0,0 +1,22 @@
+class BulkPunchesService
+  class << self
+    def call(collection_of_punches, user)
+      insert_days = collection_of_punches.map { |punch| punch['from'] }
+      raise 'There is already a punch on the same day' if there_punch_on_the_same_day?(user, insert_days)
+
+      create_new_punch(user, collection_of_punches)
+    end
+
+    private
+
+    def there_punch_on_the_same_day?(user, insert_days)
+      user.punches.by_days(insert_days).present?
+    end
+
+    def create_new_punch(user, collection_of_punches)
+      user.punches.where(
+        company: user.company
+      ).create(collection_of_punches)
```

Comment:
I'm thinking about what will happen if you send a collection without the `project_id` or `to` data. , and why are you filtering the punches with the user company, since you are getting the punches of the current user?

Reply:
Good catch, I removed that and create the punches instances before saving now, so I don't need that.